lib/liquid_voting/delegations/delegation.ex

```diff
@@ -4,19 +4,28 @@ defmodule LiquidVoting.Delegations.Delegation do
 
   alias LiquidVoting.Voting.Participant
 
+  @primary_key {:uuid, :binary_id, autogenerate: true}
+
   schema "delegations" do
     field :proposal_url, EctoFields.URL
     field :organization_uuid, Ecto.UUID
 
-    belongs_to :delegator, Participant
-    belongs_to :delegate, Participant
+    belongs_to :delegator, Participant,
+      references: :uuid,
```

Comment:
Feels like this shouldn't be needed, belongs_to ought to work out of the box with primary uuid keys. Maybe try setting `@foreign_key_type` as well as the primary key? https://hexdocs.pm/ecto/Ecto.Schema.html#module-schema-attributes

Reply:
Good idea, I saw that in the docs, but for some reason didn't try it.

Reply:
Just adding `@foreign_key_type :binary_id` and reverting to simpler:
```
belongs_to :delegator, Participant
belongs_to :delegate, Participant
```
... doesn't work. Defaults to expecting participant.id, etc.

Reply:
works fine when I add `@foreign_key_type :binary_id` and leave my more verbose, specific, 'belongs_to' statements.

Reply:
Didn't help to reduce the complexity of the need 'belong_to', but added '@foreign_key_type' in any case.

Reply:
try just keeping the `type: :binary_id` option but removing `references: :uuid` and `foreign_key: :delegator_uuid`? from what I've been seeing on ecto discussions, that should work

Reply:
Tried that (now), and get: warning: invalid association `participant` in schema LiquidVoting.Voting.Vote: associated schema LiquidVoting.Voting.Participant does not have field `id`

Reply:
ok, let me know if you don't have any luck with @niklaslong when you have a look together, else I'll go ahead and merge (resetting the production db just before it)

Reply:
We just had a look together. No concerns thus far. Niklas wants to look at the PR tomorrow morning to give it a more thorough inspection.

Reply:
sounds good 👍

Reply:
We could reduce boilerplate using macros (I wouldn't bother at this point tbh as it complicates something simple) or by naming the primary keys `id` instead of `uuid` as that's what `Ecto` defaults to.

Reply:
Yes, we could change all names that use 'uuid' (including fkeys; 'participant_uuid', for example) to 'id', and this issue would resolve. I like that we use 'uuid', however, as it highlights the use of uuids to anyone reviewing the code, db, etc. If we decide to revert to using the 'id' naming scheme, however,I can do that.